pages/[id]/index.js

```diff
@@ -0,0 +1,50 @@
+import { useRouter } from "next/router";
+import styled from "styled-components";
+import apiUrl from "next-api-url";
+
+export async function getServerSideProps(ctx) {
+  const res = await fetch(`${apiUrl(ctx)}/formdata/${ctx.query.id}`);
+  const data = await res.json();
```

Comment:
Don't fetch from your API in getServerSideProps. Read from your service / database directly instead.